lib/ansible/modules/cloud/vmware/vmware_guest_disk.py

```diff
@@ -631,9 +681,25 @@ def sanitize_disk_inputs(self):
                 self.module.fail_json(msg="Invalid 'scsi_type' specified for disk index [%s]. Please specify"
                                           " 'scsi_type' value from ['%s']" % (disk_index,
                                                                               "', '".join(self.scsi_device_type.keys())))
+
             current_disk['scsi_type'] = scsi_contrl_type
 
-            disks_data.append(current_disk)
+            # Todo some useful checks
+            if disk.get('type') == 'raw':
+                compatibility_mode = disk.get('compatibility_mode', 'physicalMode')
+
+                if compatibility_mode not in ['physicalMode']:
+                    self.module.fail_json(msg="Invalid 'compatibility_mode' specified for disk index [%s]. Please specify"
+                                          "'compatibility_mode' value from ['physicalMode']." % disk_index)
+
+                    current_disk['disk_compatibility_mode'] = compatibility_mode
+
+                if 'raw_device' not in disk:
+                    self.module.fail_json(msg="raw_device needs to be defined when using disk type 'raw' for disk index [%s]" % disk_index)
+                else:
+                    current_disk['disk_raw_device'] = disk.get('raw_device')
+
+                disks_data.append(current_disk)
```

Comment:
I tested change with my setup and looks like this line should be outside this `if` statement. This change is causing the failure in the CI test case. Could you please check this again?

Reply:
Should be resolved. Tested rdm locally.